kiliautoml/models/_hugging_face_text_classification_model.py

```diff
@@ -0,0 +1,236 @@
+
+        model_framework = set_default(
+            model_framework,
+            ModelFramework.PyTorch,
+            "model_framework",
+            [ModelFramework.PyTorch, ModelFramework.Tensorflow],
+        )
+        model_name = set_default(
+            model_name,
+            ModelName.BertBaseMultilingualCased,
+            "model_name",
+            [ModelName.BertBaseMultilingualCased],
+        )
+        return self._train(
+            assets,
+            job,
+            job_name,
+            model_framework,
+            model_name,
+            path,
+            clear_dataset_cache,
+        )
+
+    def predict(
+        self,
+        assets: List[Dict],
+        model_path: Optional[str],
+        job_name: str,
+        verbose: int = 0,
+    ) -> JobPredictions:
+
+        model_path_res, model_repository, model_framework = self._extract_model_info(
+            job_name, self.project_id, model_path
+        )
+
+        predictions = []
+        proba_assets = []
+
+        tokenizer, model = self._get_tokenizer_and_model(
+            model_framework, model_path_res, "classification"
+        )
+
+        for asset in assets:
+            response = requests.get(
+                asset["content"],
+                headers={
+                    "Authorization": f"X-API-Key: {self.api_key}",
+                },
+            )
+
+            text = response.text
+
+            predictions_asset = self._compute_asset_classification(
+                model_framework, tokenizer, model, text
+            )
+
+            predictions.append({job_name: predictions_asset})
+            proba_assets.append(predictions_asset["categories"][0]["confidence"])
+
+            if verbose:
+                print("----------")
+                print(text)
+                print(predictions_asset)
+
+        # Warning: the granularity of proba_assets is the whole document
+        job_predictions = JobPredictions(
+            job_name=job_name,
+            external_id_array=[a["externalId"] for a in assets],
+            json_response_array=predictions,
+            model_name_array=["Kili AutoML"] * len(assets),
+            predictions_probability=proba_assets,
+        )
+
+        return job_predictions
+
+    def _train(
+        self,
+        assets: List[Dict],
+        job: Dict,
+        job_name: str,
+        model_framework: str,
+        model_name: str,
+        path: str,
+        clear_dataset_cache: bool,
+    ) -> float:
+
+        kili_print(job_name)
+        path_dataset = os.path.join(path, "dataset", "data.json")
+        kili_print(f"Downloading data to {path_dataset}")
+        if os.path.exists(path_dataset) and clear_dataset_cache:
+            os.remove(path_dataset)
+        job_categories = categories_from_job(job)
+        if not os.path.exists(path_dataset):
+            with open(ensure_dir(path_dataset), "w") as handler:
+                for asset in assets:
```

Comment:
the for loop should be a dedicated function

Reply:
As a rule of thumb, more than 3 indentation level is too much?

Reply:
will be refactored